backend/de.metas.business/src/main/java/de/metas/payment/interceptor/C_Payment.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * #%L
+ * de.metas.business
+ * %%
+ * Copyright (C) 2020 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.payment.interceptor;
+
+import de.metas.adempiere.model.I_C_Invoice;
+import de.metas.allocation.api.IAllocationBL;
+import de.metas.document.DocTypeId;
+import de.metas.document.IDocTypeBL;
+import de.metas.invoice.service.IInvoiceDAO;
+import de.metas.order.IOrderDAO;
+import de.metas.order.OrderId;
+import de.metas.payment.PaymentId;
+import de.metas.payment.api.IPaymentDAO;
+import de.metas.util.Services;
+import org.adempiere.ad.modelvalidator.annotations.Interceptor;
+import org.adempiere.ad.modelvalidator.annotations.ModelChange;
+import org.compiere.model.I_C_Order;
+import org.compiere.model.I_C_Payment;
+import org.compiere.model.ModelValidator;
+import org.springframework.stereotype.Component;
+
+@Interceptor(I_C_Payment.class)
+@Component
+public class C_Payment
```

Comment:
(i wrote this comment after the two that follow in this class)

I understand that this is aiming at completed payments that get their respective invoice from your process.
* I think that your process is the only one to set C_Order_ID and C_Invoice_ID on an already completed payment
* for not-yet-completed payment, we AFAIU have Cristian's code

=> so, rather that have this model interceptor that "incidentally" allocated the payment,
make the allocation explicitly in the PaymentBL-code that also sets the C_Order_ID and C_Invoice_ID